### ISSUE TITLE
Sourcemap link is not the last thing in file.

### ISSUE THREAD

**Issue by MicahZoltu** [rollup/rollup#3148]
- Rollup Version:
- Operating System (or Browser):
- Node Version:

**Comment by MicahZoltu** [comment]
Sorry to anyone subscribed to this repository, I accidentally hit ctrl-enter which submitted the issue before filling it in.  I have updated the issue description now though.

**Comment by kzc** [comment]
> Rollup is generating a file that has a trailing newline. This MAY go against an implementation of the source map specification.

I disagree. The source map spec is informal and editors commonly add either \n or \r\n to the end of files if a source map URL is added manually. Any good source map URL reading implementation would handle the presence or absence of newlines as all browsers currently do.

**Comment by MicahZoltu** [comment]
@kzc While I tend to agree that sourcemap users should be permissive in what they accept (and I believe there is an open PR for es-module-shims to make it more permissive), I also believe that Rollup should follow the strictest interpretation of the specification to obtain maximum compatibility.  This follows the [Robustness Principle](https://en.wikipedia.org/wiki/Robustness_principle) design guideline:
> Be conservative in what you send, be liberal in what you accept

**Comment by kzc** [comment]
The source map spec provided states:

```
There is an existing convention that should be supported for the use of source maps with eval’d code, it has the following form:

//@ sourceURL=foo.js

It is described here:

http://blog.getfirebug.com/2009/08/11/give-your-eval-a-name-with-sourceurl/ 
```

It appears to be a dead link, but here's an archived copy of that web page:

https://web.archive.org/web/20190203041919/http://blog.getfirebug.com/2009/08/11/give-your-eval-a-name-with-sourceurl/

An actual code fragment with a regular expression is provided:
```
    const reURIinComment = ///@ssourceURL=s*(S*?)s*$/m;
```

I assume that the web archive dropped the necessary backslashes to make this valid JS. If one fixes the regexp you can see that optional trailing whitespace - including newlines - is permissible:
```
    const reURIinComment = /\/\/@\ssourceURL=\s*(\S*?)\s*$/m;
```
Just to back up that assertion, I see that that web page makes mention of:

https://web.archive.org/web/20190203041940/https://pmuellr.blogspot.com/2009/06/debugger-friendly.html

which in turn mentions:

https://web.archive.org/web/20161221094120/https://bugs.webkit.org/show_bug.cgi?id=25475

which provides this valid JS:

```
    // if no URL, look for "//@ sourceURL=" decorator     
    if (!sourceURL) {
        var pattern = /.*\s*\/\/\s*@\s*sourceURL\s*=\s*(\S+)\s*/;
        var match = pattern.exec(source);
        
        if (match) {
            this.sourceURL = "eval:" + match[1];
        }
    }
```

So *any* source map URI at the end of the source code with or without trailing whitespace - including newlines - are conforming according to the sourcemap spec. `es-module-shims` does not conform to this spec.
